src/__tests__/labs/requests/NewLabRequest.test.tsx

```diff
@@ -229,29 +227,14 @@ describe('New Lab Request', () => {
 
       userEvent.type(screen.getByPlaceholderText(/labs.lab.patient/i), 'Jim Bob')
 
-      await waitFor(
-        () => {
-          expect(screen.getByText(/jim bob/i)).toBeVisible()
-        },
-        { timeout: 3000 },
-      )
+      expect(await screen.findByText(/jim bob/i)).toBeVisible()
       userEvent.click(screen.getByText(/jim bob/i))
       userEvent.type(screen.getByPlaceholderText(/labs\.lab\.type/i), expectedLab.type)
       userEvent.type(screen.getByLabelText(/labs\.lab\.notes/i), (expectedLab.notes as string[])[0])
 
-      userEvent.click(screen.getByRole('button', { name: /labs\.requests\.new/i }))
-
-      await waitFor(() => {
-        expect(LabRepository.save).toHaveBeenCalledTimes(1)
+      await act(async () => {
+        userEvent.click(screen.getByRole('button', { name: /labs\.requests\.new/i }))
       })
```

Comment:
Why is act being added back in? We shouldn't need it, right?

Reply:
When I removed the expect called method it would no longer pass without an error about act not being used because a state change occurs

Reply:
did you try wrapping the `expect` in `waitFor()`?

Reply:
I can't recall if I tried that, but I'll give it another go tonight. 
So if I understand correctly, awaiting the button click basically means "clicking this will perform some kind of update, so wait for that to finish"?

Reply:
Using `waitFor` on a button click is kinda pointless - it will still only click once (unless the selector you are using to get the element fails).

Looking at the `onSave` code, it seems the issue is that the component calls `setError` (i.e. state) _after_ it has already navigated away, saved to the repository, and sent the toast notification.

```typescript
const onSave = async () => {
    try {
      const newLab = await mutate(newLabRequest as Lab)
      history.push(`/labs/${newLab?.id}`)
      Toast(
        'success',
        t('states.success'),
        `${t('labs.successfullyCreated')} ${newLab?.type} ${t('labs.lab.for')} ${
          (await PatientRepository.find(newLab?.patient || '')).fullName
        }`,
      )
      setError(undefined)
    } catch (e) {
      setError(e)
    }
  }
```

Reply:
I was saying use the `waitFor` on the `expect`

Reply:
Sooo.. is `waitFor` the right way to go in this case because we don't want to expect a `Called` method?

Reply:
`waitFor` is already being done on the expect for the `history` change... I wonder if rendering the Toaster component, and checking for the Toast would help (and make for a better test)?

Reply:
Mm gotcha. I will try out adding the Toaster